tests/autorag/evaluate/metric/test_retrieval_metric.py

```diff
@@ -19,26 +20,41 @@
     ['test-13', 'test-12', 'pred-10', 'pred-11'],  # recall: 2/3, precision: 0.5, f1: 4/7
     ['test-14', 'pred-12'],  # recall: 1.0, precision: 0.5, f1: 2/3
     ['pred-13'],  # retrieval_gt is empty so not counted
-    ['pred-14']  # retrieval_gt is empty so not counted
+    ['pred-14'],  # retrieval_gt is empty so not counted
+    ['pred-15', 'pred-16', 'test-15']  # recall:1, precision: 1/3, f1: 0.5
 ]
 
 
 def test_retrieval_f1():
-    solution = [0.5, 2 / 7, 2 / 5, 4 / 7, 2 / 3, None, None]
+    solution = [0.5, 2 / 7, 2 / 5, 4 / 7, 2 / 3, None, None, 0.5]
     result = retrieval_f1(retrieval_gt=retrieval_gt, pred_ids=pred)
     for gt, res in zip(solution, result):
         assert gt == pytest.approx(res, rel=1e-4)
 
 
 def test_retrieval_recall():
-    solution = [0.5, 1 / 3, 1, 2 / 3, 1, None, None]
+    solution = [0.5, 1 / 3, 1, 2 / 3, 1, None, None, 1]
     result = retrieval_recall(retrieval_gt=retrieval_gt, pred_ids=pred)
     for gt, res in zip(solution, result):
         assert gt == pytest.approx(res, rel=1e-4)
 
 
 def test_retrieval_precision():
-    solution = [0.5, 0.25, 0.25, 0.5, 0.5, None, None]
+    solution = [0.5, 0.25, 0.25, 0.5, 0.5, None, None, 1 / 3]
     result = retrieval_precision(retrieval_gt=retrieval_gt, pred_ids=pred)
     for gt, res in zip(solution, result):
         assert gt == pytest.approx(res, rel=1e-4)
+
+
+def test_retrieval_ndcg():
+    solution = [0.9197207891481876, 1, 1, 1, 1, None, None, 0.5]
+    result = retrieval_ndcg(retrieval_gt=retrieval_gt, pred_ids=pred)
+    for gt, res in zip(solution, result):
+        assert gt == pytest.approx(res, rel=1e-4)
+
+
+def test_retrieval_mrr():
+    solution = [1, 1, 1, 1, 1, None, None, 1 / 3]
```

Comment:
This is also weird result.
I think the first case result must be 0.5. 
There is two result set, and the first one appeared at the first rank.
Its reciprocal Rank (RR) will be 1. `test-2` in third rank will be ignored because you already found `test-1`. But there is no `test-3` in the pred set. There is two answer set, so the final mRR will be (1/2) * 1 => 0.5

Your code might wrong.